baseapp/baseapp.go

```diff
@@ -610,7 +610,6 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (gInfo sdk.
 		gasWanted = ctx.GasMeter().Limit()
 
 		if err != nil {
-			gInfo = sdk.GasInfo{GasWanted: gasWanted, GasUsed: ctx.GasMeter().GasConsumed()}
 			return gInfo, nil, err
```

Comment:
nit: A simple `return` would suffice here correct?

Reply:
you mean a naked return? Yes, but that drastically reduces comprehensibility.  Naked returns should only be used in very short/concise methods & functions.